engine/src/witness/sol/fee_tracking.rs

```diff
@@ -0,0 +1,24 @@
+use crate::sol::{
+	retry_rpc::{SolRetryRpcApi, SolRetryRpcClient},
+	rpc_client_api::RpcPrioritizationFee,
+};
+
+pub async fn get_median_prioritization_fee(sol_client: &SolRetryRpcClient) -> Option<u64> {
+	let mut fees = sol_client
+		.get_recent_prioritization_fees()
+		.await
+		.into_iter()
+		.map(|RpcPrioritizationFee { prioritization_fee, .. }| prioritization_fee)
+		.collect::<Vec<_>>();
```

Comment:
We can do this after checking if empty

Reply:
(And IMO reads a little better like this)